app/src/main/java/http_server/ResponseBuilder.java

```diff
@@ -96,6 +104,17 @@ private Response handleRedirectResponse(String method) {
     }
     return handler.buildUnexpectedResponse("405 Method Not Allowed");
   }
+
+  Response handleCreateResourceResponse(String method, String body) throws IOException {
+
+    if (method.equals("POST") && body != null) {
+      fileCreator.createFile(body);
+
+      return handler.buildResourceResponse("Content-Type", "application/json;charset=utf-8", "201 Created", body);
+    }
+    return handler.buildUnexpectedResponse("405 Method Not Allowed");
+  }
+
```

Comment:
It is nice to see that you just needed to add another case and the pattern still works.